homeassistant/components/homematic/__init__.py

```diff
@@ -139,6 +139,7 @@
                     cv.match_all: {
                         vol.Required(CONF_HOST): cv.string,
                         vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+                        vol.Optional(CONF_JSONPORT, default=DEFAULT_JSONPORT): cv.port,
```

Comment:
We no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

This integration needs to be refactored to use a config flow and config entries.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler

Reply:
Ok, thank you for the hint, I hoped bugfixes are OK. I am afraid my programming skills will not be enogh for that big change. So, you can finallly decline this PR.

Reply:
Sorry to hear that @wladwnt. If you change your mind and want to give it a shot, feel free to re-open a PR for it 👍 (and you are, of course, always welcome on our Discord server for some developer assistance).

Reply:
I spent in total about 8-10 hours to make this PR (it my first one, so had to understand some basic things), and it is completly demotivating if a small bugfix PR (indeed only two lines) is declined because of the rule, that is, in my opinion, not well-thought-out. Why not to accept small bugfixes to yaml configuration, even if it is deprecated in this context. That force me rather to do changes I need locally than try to contribute to HA again. Unfortunately.

Reply:
I understand your frustration, however, there has to be a line drawn somewhere. This is why we have those Architectural Decision Records (ADR) as linked above.